src/ts/deploy.ts

```diff
@@ -31,18 +46,30 @@ const DEPLOYER_CONTRACT = "0x4e59b44847b379578588920ca78fbf26c0b4956c";
  * @param deploymentArguments Extra arguments that are necessary to deploy.
  * @returns The address that is expected to store the deployed code.
  */
-export async function deterministicDeploymentAddress(
-  contractName: string,
-  ...deploymentArguments: unknown[]
+export async function deterministicDeploymentAddress<C extends ContractName>(
+  contractName: C,
+  ...deploymentArguments: DeploymentArguments<C>
 ): Promise<string> {
-  const factory = await ethers.getContractFactory(contractName);
-  const deployTransaction = factory.getDeployTransaction(
-    ...deploymentArguments,
-  );
+  // NOTE: Use `require` to load the contract artifact instead of `getContract`
+  // so that we don't need to depend on `hardhat` when using this project as
+  // a dependency.
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const { abi, bytecode } = require(getArtifactPath(contractName));
```

Comment:
Why `require()` and not [dynamic `import()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports)?

Reply:
Because I didn't know it existed!